CMakeLists.txt

```diff
@@ -0,0 +1,53 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+cmake_minimum_required(VERSION 3.20)
```

Comment:
Is there any reason for CMake 3.20? In Arrow C++ there was a discussion lately to try and move to CMake 3.25 to allow using some of the newer feature like:
- `OVERRIDE_FIND_PACKAGE` added to `FetchContent` on 3.24
- file_sets for easier header management and installation added on 3.23 (https://cmake.org/cmake/help/latest/command/target_sources.html#file-sets)

CMake is easily installable but the default versions on newer distributuions are:
- Debian GNU/Linux bookworm provides CMake 3.25: https://packages.debian.org/search?keywords=cmake
- Ubuntu 24.04 provides CMake 3.28: https://packages.ubuntu.com/search?keywords=cmake
- AlmaLinux 8 provides CMake 3.26
- AlmaLinux 9 provides CMake 3.26